base/broadcast.jl

```diff
@@ -280,9 +280,21 @@ Base.@propagate_inbounds function Base.iterate(bc::Broadcasted, s)
 end
 
 Base.IteratorSize(::Type{T}) where {T<:Broadcasted} = Base.HasShape{ndims(T)}()
-Base.ndims(BC::Type{<:Broadcasted{<:Any,Nothing}}) = _maxndims(fieldtype(BC, :args))
-Base.ndims(::Type{<:Broadcasted{<:AbstractArrayStyle{N},Nothing}}) where {N<:Integer} = N
+Base.ndims(BC::Type{<:Broadcasted{<:Any,Nothing}}) = _maxndims_broadcasted(BC)
+function Base.ndims(BC::Type{<:Broadcasted{<:AbstractArrayStyle{N},Nothing}}) where {N}
+    if (N isa Type) && (Any <: N)
```

Comment:
Are there any types for which this is not the same as `N === Any`?

Reply:
Switched to using dispatch instead of an `if` expression, so the question is not relevant any more.

Reply:
I'm still generally curious about the answer to that question if you know it, even if it isn't relevant to the PR.

Reply:
Found one :smile:

```julia-repl
julia> t = Union{A, B} where {A>:Integer, B>:AbstractFloat}
Union{A, B} where {A>:Integer, B>:AbstractFloat}

julia> Any <: t
true

julia> Any === t
false
```

Reply:
That said, Julia *is* able to normalize the above `UnionAll` to `Any` when the type variables are made inaccessible for type application:

```julia-repl
julia> t = Union{A, B} where {A>:Integer, B>:AbstractFloat}
Union{A, B} where {A>:Integer, B>:AbstractFloat}

julia> Val{t}
Val{Any}
```